### PR TITLE
perf: added error Handling for homepage component

### PR DESCRIPTION
**Pull request by HarshMN2345**
It introduces error handling for useCurrentUser errors, adds a loading state, implements an authentication redirect, and improves code organization.

### CODE REVIEW

File: pages/index.tsx [hunk 1]
```diff
@@ -7,14 +7,27 @@ import { Button } from '@/components/ui/button'
...
-    if (!user) {
+    if (error) {
+      setErrorMessage('An error occurred while fetching user data. Please try again later.')
+    }
+
+    if (user && !isLoading) {
+      router.replace('/dashboard')
+    }
+
+    if (!user && !isLoading) {
```
Comment (piyushgarg-dev) [inline pages/index.tsx:23]: Hi @HarshMN2345
Thanks for contributing 😄.

This is the homepage as we would like a Landing page here. We don't want to redirect the user to `/dashboard`.

Also, the dashboard is at `app.localhost:3000/dashboard` subdomain 😄

But great code 🎉

**Comment by piyushgarg-dev** [comment]
Also please mention #issue-number in description ✨

**Comment by piyushgarg-dev** [comment]
@HarshMN2345 Please update the issue number, You have mentioned wrong issue number

**Review by hemantwasthere — COMMENTED** [review]
There are so many bugs in the code so closing the PR

**Pull request closed without merging**